### DISCUSSION TITLE
Comunicação entre times

### DISCUSSION THREAD

**Discussion by henriquecbuss** [cambiatus/frontend#753] [General]
## Contexto

Hoje, usamos 3 ferramentas principais para nos comunicarmos dentro da Cambiatus:

- Slack
- Figma
- GitHub

Não temos nenhum "acordo formal" sobre o que deve ser comunicado em cada ferramenta, e como devemos realizar a comunicação entre diferentes times. Deste modo, muitas vezes acabamos tendo pedaços de informações espalhados entre todas as ferramentas, fazendo com que nenhuma delas sozinha tenha toda a informação.

Durante nossas reuniões semanais do time de dev, passamos pelas issues que temos em aberto no GitHub, e verificamos o estado de cada uma (se falta algo do time de ui/ux, algum input do time de community building, etc.). Muitas vezes, só olhando pelos comentários da issue, não conseguimos entender o estado dela (como foi o caso de #660, que incentivou a criação dessa discussão), e precisamos conferir com os outros times para ver se está tudo certo.

---

Para o time de dev, seria ótimo se pudéssemos ter todo o contexto necessário aqui no GitHub. Sempre que vamos resolver uma issue, temos que lê-la primeiro, e podemos entender o contexto e possíveis mudanças no texto original da issue a partir dos comentários.

- Aqui no GitHub, fica tudo documentado e aberto para qualquer um ver (o que facilita possíveis colaboradores de fora da Cambiatus). 
- Como já passamos pelo texto das issues de qualquer forma, fica bem mais fácil encontrarmos o que precisamos
- É muito mais fácil procurarmos por informações aqui no GitHub do que no Slack ou Figma

Para o time de design, é muito mais conveniente ter um comentário no Figma, exatamente no local que o comentário menciona. Porém, seria bom se tivéssemos um resumo (ou o link) do que foi decidido no Figma postado no GitHub também, para nos mantermos atualizados sempre, e não correr o risco de passar batido na hora de resolver a issue.

Em relação ao que não for específico a uma issue, podemos usar a funcionalidade de [Discussions](https://github.com/orgs/cambiatus/discussions) do GitHub, que é um ambiente mais aberto, e feito justamente para discutirmos coisas de mais alto nível

---

Seria legal termos pontos de vistas de outros times também @lucca65 @juramos-2020 @rafachadud @muguika para decidirmos o que é melhor para cada um 😄

**Comment by henriquecbuss** [comment]
Para não perdermos o que já foi falado em #660 , aqui estão os comentários:

- @rafachadud 

> Tranquilo, eu mesma me perco no GitHub e no Figma todo o tempo 🫠
A @juramos-2020 está me ajudando e me relembrando o que ficou para trás. 🫶

> Talvez possamos começar a discussão e feedbacks no slack, e colocar aqui quando a feature/interface estiver refinada, e usar o github em caso de precisarmos de algum update?


- @juramos-2020 

> Acho que quando for uma discursão mais ampla, principalmente que possa ter envolvimento da Karla e do Luiz, seria interessante começarmos a discursão no slack (canal de dev ou ux-ui).

> Pra mim o figma funciona bem quando é para os Devs indicarem alguma alteração simples na interface, algo bem pontual. Mas se quiserem compartilhar o link da tela por aqui e descrever a alteração, também funciona 😊

**Comment by lucca65** [comment]
Concordo. Acredito que no Github hub, só por ser aberto, temos a chance de ser muito mais ricos. O fato de ser mais trabalhoso também nos força a dar mais atenção ao escrever aqui. 

Fora que o GitHub tem ferramentas de notificação e histórico. +1 mas depende muito da @rafachadud e da @juramos-2020 pq acho que elas são as mais afetadas pela mudança/unificação da comunicação

**Comment by juramos-2020** [comment]
Olá pessoal,

Por mim podemos continuar usando o Github para seguir com as discussões e as atualizações tranquilo, não precisamos mudar isso não 😅 

Acho que podemos só alinhar e nos ajudar em alguns sentidos, ex:
- se fizermos uma pergunta no figma, será respondido no figma
- se fizermos uma pergunta no slack, será respondido no slack
- se fizermos uma pergunta no git será respondido nele.

E claro, sempre usando o "add link" que é um grande aliado nessa construção da comunicação 😊

**Minha sugestão é de usarmos os links** para podermos usar essas tres ferramentas (figma, git e slack), pois cada time trabalha com uma e quem não trabalha com elas usa somente o slack rsrsrs

Acho que me equivoquei em dizer "começarmos a discussão no slack". Na verdade não seria começar e sim **"Compartilhar a discussão no slack"**

Quando eu falo sobre compartilhar no slack, é porque dependendo da discussão, principalmente quando envolvem mudanças mais profundas de funcionalidade e propósito (porque aquilo está sendo criado), se tratamos dessa discussão somente aqui no Github acabamos limitando a conversa entre os Devs e Designers e excluindo outras opiniões que poderiam agregar muito na discussão. Muitos membros da Cambiatus não usam o Github, apenas o slack.

Quando sugiro de levar essas discussões para o slack, isso não impede que ela seja iniciada e postada aqui no GitHub. Mas quem levantou a discussão ou está pedindo uma alteração de interface, essa pessoa pode pegar o link do comentário (que foi feito aqui no Gthub) ou da Issue completa e compartilhar no slack para que mais pessoas tenham acesso à ela. 

Quando eu falo sobre mudanças pontuais nas interfaces, me refiro a alterações simples, ex: adiciona um espaço aqui, coloca um componente alí, acho que aqui deveria ser roxo.... essas coisas simples. Vocês podem, e devem solicitar essa alteração por aqui (pelo Github) sim, pois como o Lucca disse, aqui fica tudo registrado. Mas mais uma vez podemos usar os links ao nosso favor. Quem fizer a solicitação pode fazer direto no figma, e copiar a mesma solicitação aqui no GitHub adicionando o link do figma (o link pode ser do comentário ou da tela). Isso ajuda pois faz a gente ganhar tempo. 😁💚

**Reply by lucca65** [reply]
Entendi, gostei muito @juramos-2020! Responder na ferramenta mas ter um lugar central pra ficar fácil de manter a conversa documentada. 

Nessa issue que gerou essa discussão que estamos inclusive, tinha o link, a @rafachadud achou e colou lá. 

Acha que seria interessante a gente colocar isso em algum lugar? Talvez salvar aqui na discussão de maneira resumida?

**Comment by juramos-2020** [comment]
> Responder na ferramenta mas ter um lugar central pra ficar fácil de manter a conversa documentada.

sim 😉👍

> Acha que seria interessante a gente colocar isso em algum lugar? Talvez salvar aqui na discussão de maneira resumida?

Desculpa Lucca não entendi 🙈 O que você gostaria de fazer?

**Reply by lucca65** [reply]
@juramos-2020 acho que lendo ontem eu entendi errado sua proposta. Acho tudo bem a gente usar o slack como ferramenta preferencial para discutir mas o problema todo é que não temos um lugar pra referencia do que foi decidido. Principalmente no slack é muito difícil lembrar de uma conversa que tivemos depois de muito tempo. Aqui no Github é fácil de retomar oq foi discutido mesmo depois de anos.

O que acha dessa proposta:

> Usamos links para referenciar o que for necessário, mas as decisões precisam ficar no GitHub, para que saibamos disso ao implementar as funcionalidades, sem esquecer de nenhuma decisão.

O @NeoVier tem feito isso várias vezes, mas acho que podemos nós 3 (eu, @juramos-2020 e @rafachadud) também passar a fazer o mesmo

**Reply by juramos-2020** [reply]
Oi Lucca 

A minha proposta era que usemos os links. 

Não estou propondo que a gente use o slack como ferramenta preferencial. A proposta é que a conversa se inicie no Github e que o link seja compartilhado no slack quando for o caso de precisar de outras opiniões de outros membros.

**Reply by juramos-2020** [reply]
Sobre o uso do link eu tenho utilizado bastante também, então a idéia é só seguir usando 😉👍

**Reply by juramos-2020** [reply]
O lugar central que ficaria tudo registrado seria justamente aqui.

**Reply by lucca65** [reply]
Ah legal! Então beleza. Acho que podemos fechar então: GitHub fica como "referência das nossas decisões". O que acham @NeoVier e @rafachadud ?

**Reply by iamluizhadad** [reply]
Adorei essa discussao. Inclusive ela vai me motivar a usar mais o GitHub. Acredito que podemos fazer o onboarding do time de Community Building aqui, e acredito que compartilhar links sobre as issues/pedidos de colaboracao dos membros no slack/outros canais pra trazer a discussao pra ca como "referencia das decisoes" e um bom acordo :)

**Reply by lucca65** [reply]
Incrível! valeu @iamluizhadad
